### PR TITLE
Sample: Compare similarity using embeddings

### PR DESCRIPTION
**Pull request by oshadajay**
## Purpose
Add ballerina sample for comparing similarity between two texts using GPT3 embeddings.

### CODE REVIEW

File: compare_similarity_using_embeddings/main.bal [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import ballerina/io;
+import ballerinax/openai;
+import ballerina/lang.'float as langFloat;
+
+configurable string openAIKey = ?;
+
+function vectorNorm(float[] vector1, string norm = "l2") returns float|error {
```
Comment (malithjayasinghe) [inline compare_similarity_using_embeddings/main.bal:7]: Why do you have these functions here? Import the math library and use functions in the math library

**Review by malithjayasinghe — CHANGES_REQUESTED** [review]

File: compare_similarity_using_embeddings/main.bal [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        float dot_product = 0.0;
+        foreach int i in 0 ..< vector1.length(){
+            dot_product += vector1[i] * vector2[i];
+        }
+        return dot_product;
+    }
+}
+
+function cosineSimilarity(float[] vector1, float[] vector2) returns float|error {
+    // Calculate cosine similarity between two vectors
```
Comment (nadheesh) [inline compare_similarity_using_embeddings/main.bal:45]: example should use the library

File: compare_similarity_using_embeddings/main.bal [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    openai:CreateEmbeddingRequest text1_embedding_request = {
+        model: "text-embedding-ada-002",
+        input: text1
+    }; 
+
+    openai:CreateEmbeddingRequest text2_embedding_request = {
+        model: "text-embedding-ada-002",
+        input: text2
+    };
```
Comment (nadheesh) [inline compare_similarity_using_embeddings/main.bal:77]: can't we do a batch call?

File: compare_similarity_using_embeddings/main.bal (hunk 2 above)
Comment (oshadajay) [reply compare_similarity_using_embeddings/main.bal]: These functions are temporarily added.

File: compare_similarity_using_embeddings/main.bal (hunk 1 above)
Comment (oshadajay) [reply compare_similarity_using_embeddings/main.bal]: These functions are temporarily added.

**Review by malithjayasinghe — APPROVED** [review]

**Pull request merged**